### ISSUE TITLE
[rom_ctrl, dv] Uncovered conditional coverage in rom_ctrl_fsm

### ISSUE THREAD

**Issue by KinzaQamar** [lowRISC/opentitan#24720]
The nightly regression coverage dashboard reports an uncovered conditional coverage inside rom_ctrl_fsm.sv [here](https://github.com/lowRISC/opentitan/blob/6d8b4077d588ba1315770a5749e4036dcb51980a/hw/ip/rom_ctrl/rtl/rom_ctrl_fsm.sv#L148).
The case 110 is uncovered. If ```counter_lnt``` is true, it means that we've read the last non-top item from ROM which basically implies that ```kmac_rom_vld_o``` is also high. That is a non-occurring case.

**Comment by rswarbrick** [comment]
I agree and *think* that this isn't actually reachable. My reasoning: the `kmac_rom_vld_o` signal basically says "we read something from ROM on the last cycle". If `counter_lnt` is true, it means "we've just read the last non-top item from ROM" and this implies that `kmac_rom_vld_o` is high.

One option would be a waiver (possibly UNR), but I think it's probably cleaner to just drop the condition from the RTL. Ideally, by running UNR or a manual FPV flow to prove that the change is correct. I'll take this on.

**Comment by KinzaQamar** [comment]
Resolved [25682](https://github.com/lowRISC/opentitan/pull/25682). Thanks!
